R/irace_helpers.R

```diff
@@ -0,0 +1,64 @@
+paradoxToIrace = function(ps){
+  assertClass(ps, "ParamSet")
+  # what about ParamUty = vector numeric/real
+  class_lookup = data.frame(paradox = c("ParamLgl","ParamInt","ParamDbl","ParamFct"),
+                            irace = c("c","i","r","c"), stringsAsFactors = FALSE)
+  name <- ps$ids()
+  type <- class_lookup[match(ps$class, class_lookup$paradox), 2]
+  range <- getIraceRange(ps)
+  if(ps$has_deps){
+    condition <- getIraceCondition(ps)
+  } else {
+    condition = NULL
+  }
+
+  # does mlr3tuning deal with trafo internally or passed to irace? irace only handle log
+  par.tab <- paste(name, '""', type, range, condition, collapse = "\n")
+  irace::readParameters(text = par.tab)
+}
+getIraceRange = function(ps){
+  rng = data.table(lower = ps$lower, upper = ps$upper, lvl = ps$levels)
```

Comment:
`as.data.table()` should be enough

Reply:
What do you mean by enough?

Reply:
you can probably do something with
```r
as.data.table(ps)[, ifelse(is.na(lower),
  <whatever paste collapse 'levels'>,
  paste0("(", lower, ",", upper, ")"))]
```

Reply:
Do you know the best way to collapse lists with some elements characters and some not? 

```R
as.data.table(ps)[,ifelse(is.na(lower),
                            paste0("(",paste0(levels, collapse = ","),")"),
                            paste0("(",lower,",",upper,")"))]
```

Doesn't work as it collapses all `levels` and ignores the `ifelse, e.g. I got:

```
"c(TRUE, FALSE),NULL,c(\"lvl1\", \"lvl2\"),NULL" "(1,9)" 
```

Reply:
add `by = id` and use `levels[[1]]`:
```r
as.data.table(ps(x = p_dbl(1, 4), y = p_lgl()))[,
  ifelse(is.na(lower),
    paste0("(",paste0(levels[[1]], collapse = ","),")"),
    paste0("(",lower,",",upper,")")),
  by = "id"]$V1
#> [1] "(1,4)"        "(TRUE,FALSE)"
```